src/CMakeLists.txt

```diff
@@ -15,7 +15,7 @@ endif()
 find_package(HDF5 REQUIRED)
 
 # std::filesystem (required for GCC < 9)
-if (NOT CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
+if ((NOT CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang") AND (NOT CMAKE_CXX_COMPILER_ID STREQUAL "Clang"))
```

Comment:
Unfortunately, this won't work. On systems that use the GNU C++ standard library version < 9 but where the user is building with Clang, it will not be able to use `std::filesystem` without linking with `stdc++fs`. This will affect all users of RHEL 8, which is currently the Linux version used on Gadi and many other clusters.